File: src/Entity/Reclamation.php
<?php

namespace App\Entity;

use App\Repository\ReclamationRepository;
use Doctrine\ORM\Mapping as ORM;


#[ORM\Entity(repositoryClass: ReclamationRepository::class)]

class Reclamation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private $id;

   
    #[ORM\Column(name: 'cin', type: 'string', length: 255, nullable: true)]
    private $cin;

    
    #[ORM\Column(name: 'type', type: 'string', length: 255, nullable: true)]
    private $type;

    
    #[ORM\Column(name: 'objet', type: 'string', length: 255, nullable: true)]
    private $objet;

   
    #[ORM\Column(name: 'message', type: 'string', length: 255, nullable: true)]
    private $message;

    #[ORM\Column(name: 'date', type: 'date', length: 255, nullable: true)]
    private $date;
     /**
     * @ORM\ManyToOne(targetEntity="App\Entity\Reponse")
     * @ORM\JoinColumn(name="id", referencedColumnName="id")
     */
    private $reponse;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCin(): ?string
    {
        return $this->cin;
    }

    public function setCin(string $cin): self
    {
        $this->cin = $cin;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getObjet(): ?string
    {
        return $this->objet;
    }

    public function setObjet(string $objet): self
    {
        $this->objet = $objet;

        return $this;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(string $message): self
    {
        $this->message = $message;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }
    public function getReponse(): ?Reponse
    {
        return $this->reponse;
    }

    public function setReponse(Reponse $reponse): self
    {
        $this->reponse = $reponse;

        // set the owning side of the relation if necessary
        if ($reponse->getReclamation() !== $this) {
            $reponse->setReclamation($this);
        }

        return $this;
    }
}

File: src/tn/gestion/promotion/enitite/Reclamation.java
    package tn.gestion.promotion.enitite;

public class Reclamation {
    private int id;
    private String cin ;
    private String type;
    private String objet;
    private String message;
    private String date;

    public String getCin() {
        return cin;
    }

    public void setCin(String cin) {
        this.cin = cin;
    }
    
    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getObjet() {
        return objet;
    }

    public void setObjet(String objet) {
        this.objet = objet;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    
    
    public Reclamation() {
    }

    public Reclamation(int id, String cin, String type, String objet, String message, String date) {
        this.id = id;
        this.cin = cin;
        this.type = type;
        this.objet = objet;
        this.message = message;
        this.date = date;
    }

    public Reclamation(String cin, String type, String objet, String message, String date) {
        this.cin = cin;
        this.type = type;
        this.objet = objet;
        this.message = message;
        this.date = date;
    }
    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }


    @Override
    public String toString() {
        return "Categorie{" + "id=" + id + ", name=" + type + ", type=" + cin + '}';
    }

}
